packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -421,6 +424,33 @@ function resolve(
 
     if (shouldConstruct(Component)) {
       inst = new Component(element.props, publicContext, updater);
+
+      if (typeof Component.getDerivedStateFromProps === 'function') {
+        partialState = Component.getDerivedStateFromProps(
+          element.props,
+          inst.state,
```

Comment:
This can be `null`, right? Do we want to pass `{}` in this case for type safety?

I don't remember what this debate settled on.

Reply:
It can be `null`, yup. I assume this was desirable since it can be used to detect that the state has not yet been initialized. Alternately if you've initialized the state via the constructor it will be whatever that value us.